src/os/bluestore/BlueStore.cc

```diff
@@ -3189,7 +3192,8 @@ int BlueStore::_read_whole_blob(const bluestore_blob_t* blob, OnodeRef o, buffer
 
   result->clear();
 
-  uint32_t l = blob->get_payload_length();
+
+  uint32_t l = blob->get_aligned_payload_length(block_size);
```

Comment:
can you separate out the compression-related accounting changes into a separate commit, so that the first commit just does the switchover from statfs to store_statfs_t, and the next commit makes bluestore smarter to fill in extra fields?